src/shared/sam/localLambdaRunner.ts

```diff
@@ -374,6 +374,11 @@ export async function runLambdaFunction(
 
     await onAfterBuild()
 
+    // If the build successfully but expires the timer, reset the timer
+    if (timer.completed) {
+        timer.reset()
+    }
```

Comment:
Why check `timer.completed`? Don't we always unconditionally want to extend the timer?

Reply:
The only time a reset is required is when the timer has completed. There are refresh calls during the invoke process that will restart any non-completed timeout.